### PR TITLE
dev-ml/cmdliner: add 1.1.1

### PR DESCRIPTION
**Pull request by Alessandro-Barbieri**
Closes: https://bugs.gentoo.org/836972
Signed-off-by: Alessandro Barbieri <lssndrbarbieri@gmail.com>

### CODE REVIEW

**Comment by xgqt** [comment]
Current opam-core fails to build with this version

```
Error: No implementations provided for the following modules:
         Cmdliner_term_deprecated referenced from /usr/lib64/ocaml/cmdliner/cmdliner.cmxa(Cmdliner)
```

**Comment by Alessandro-Barbieri** [comment]
Should I report a bug to upstream?

File: dev-ml/cmdliner/cmdliner-1.1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+RESTRICT="!test? ( test )"
+
+src_compile() {
+	ocaml pkg/pkg.ml build \
+		--tests $(usex test true false) \
+		|| die
+}
+
+src_test() {
+	ocaml pkg/pkg.ml test || die
```
Comment (dra27) [inline dev-ml/cmdliner/cmdliner-1.1.1.ebuild:35]: This isn't the correct build command - see `Makefile` in the sources, you want
```sh
ocaml build.ml cma
ocaml build.ml cmxa
ocaml build.ml cmxs
```
(not sure if they can be combined)

File: dev-ml/cmdliner/cmdliner-1.1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+IUSE="+ocamlopt test"
+
+RDEPEND="
+	>=dev-lang/ocaml-4.08.00:=[ocamlopt?]
+	dev-ml/result:=
+	dev-ml/findlib:=
+"
+DEPEND="${RDEPEND}
+	dev-ml/topkg
+	dev-ml/ocamlbuild"
```
Comment (dra27) [inline dev-ml/cmdliner/cmdliner-1.1.1.ebuild:24]: These dependencies are no longer required, I believe

**Comment by xgqt** [comment]
@Alessandro-Barbieri have you checked if it builds correctly with `opam-core`?

File: dev-ml/cmdliner/cmdliner-1.1.1.ebuild (hunk 1 above)
Comment (xgqt) [reply dev-ml/cmdliner/cmdliner-1.1.1.ebuild:37]: @Alessandro-Barbieri actually delegating this work to makefile would be better imo.

consider:
```bash
src_compile() {
    emake build-byte

    if use ocamlopt ; then
        emake build-native build-native-dynlink
    fi
}
```

**Pull request closed without merging**